DxDispatch/src/dxdispatch/dxDispatchWrapper.cpp

```diff
@@ -0,0 +1,304 @@
+#include "pch.h"
+#include "Adapter.h"
+#include "Device.h"
+#include "Model.h"
+#include "Dispatchable.h"
+#include "JsonParsers.h"
+#include "Executor.h"
+#include "CommandLineArgs.h"
+#include "ModuleInfo.h"
+#include "dxDispatchWrapper.h"
+
+using namespace Microsoft::WRL;
+
+class ModelWrapper
+{
+public:
+    ModelWrapper(Model&& model):
+        m_model(std::move(model))
+    {}
+
+    ~ModelWrapper() = default;
+    Model &Value()
+    {
+        return m_model;
+    }
+private:
+    Model m_model;
+};
+
+HRESULT DxDispatch::CreateDxDispatchFromJsonString(
+    _In_            int argc,
+    _In_            char** argv,
+    _In_opt_        LPCSTR jsonConfig,
+    _In_opt_        IUnknown *adapterUnk,
+    _In_opt_        IDxDispatchLogger *customLogger,
+    _COM_Outptr_    IDxDispatch **dxDispatch)
+{
+    ComPtr<DxDispatch> dxDispatchImpl;
+    ComPtr<IAdapter> adapter;
+
+    if(adapterUnk)
+    {
+        RETURN_IF_FAILED(adapterUnk->QueryInterface(IID_PPV_ARGS(&adapter)));
+    }
+    dxDispatchImpl = new DxDispatch();
+    RETURN_IF_FAILED(dxDispatchImpl->Intialize(argc, argv, jsonConfig, adapter.Get(), customLogger));
+    return dxDispatchImpl->QueryInterface(IID_PPV_ARGS(dxDispatch));
+}
+
+HRESULT DxDispatch::QueryInterface( 
+    REFIID riid,
+    _COM_Outptr_ void** ppvObject)
+{
+    if((IID_IUnknown ==  riid) || 
+        __uuidof(IDxDispatch) == riid)
+    {
+        *ppvObject = static_cast<IDxDispatch*>(this);
+    }
+    else
+    {
+        return E_NOTIMPL;
```

Comment:
I think this is supposed to be E_NOINTERFACE, but again it might be better to just use WRL::Base.